pymc/distributions/continuous.py

```diff
@@ -1901,7 +1901,7 @@ class StudentT(Continuous):
     rv_op = studentt
 
     @classmethod
-    def dist(cls, nu, mu=0, lam=None, sigma=None, *args, **kwargs):
+    def dist(cls, nu=1, mu=0, sigma=None, lam=None, *args, **kwargs):
```

Comment:
I think this distribution is critical enough that we need to do this non silently. The easiest way I can think of is to temporary force it to be a keyword argument, and later on we can drop that requirement again:

```suggestion
    def dist(cls, nu, mu=0, *, sigma=None, lam=None, *args, **kwargs):
```

Reply:
I think this bumps into @michaelosthege's [comment](https://github.com/pymc-devs/pymc/issues/5602#issuecomment-1071238984) about why we are accepting positional arguments at all.  Given that most of the tests are passing (which surprised me), it doesn't seem like we are expecting users to rely on this behavior.

Reply:
> I have not seen non-explicit distr parametrization in the wild.

I have seen them in code posted to discourse before.

Reply:
somewhat related, there are also several cases where `param=None` doesn't mean that parameter is optional. I think this only makes sense when combining positional and keywork args, for example things like this: https://github.com/pymc-devs/pymc/pull/5595#discussion_r826996478